rasa/core/channels/socketio.py

```diff
@@ -251,8 +255,15 @@ async def handle_message(sid: Text, data: Dict) -> None:
             else:
                 sender_id = sid
 
+            metadata = data.get(self.metadata_key, {})
+            if metadata:
+                metadata = ast.literal_eval(data.get(self.metadata_key, {}))
```

Comment:
```suggestion
                metadata = ast.literal_eval(metadata)
```
Suggesting to replace calling the expression twice with the `metadata` variable.
Also never used `ast` before, is that to make sure metadata type is one of the built-in Python data types?

Reply:
Yes, ast.literal_eval raises an exception if the input isn't a valid Python datatype

Reply:
Just to make sure I get this right, do we always expect metadata if it is retrieved from `data` dict to be of type string?
Only concerned this might throw a ValueError if metadata is a non-empty dict or list, since `ast.literal_eval` expects an argument of type string or expression node (I am not sure what the latter is actually).

Reply:
Hey @rgstephens This was actually pointed out to me today that `ast.literal_eval` can present a security risk since it can run any arbitrary code, so best to replace it with `json.loads()`